### PR TITLE
added username and password caching in login

### PR DESCRIPTION
**Pull request by marouane97**

### CODE REVIEW

**Comment by NealSun96** [comment]
@marouane97 The design principle of password cookie is to not store the password directly to any machine, but instead store the api_key. Let's modify it so that the api_key is stored and retreived from the window_storage at the start of the app. If the api_key is none, then we let users log in. 

But nice job man 👌

**Comment by marouane97** [comment]
the key is $rootScope.api_auth right?

**Comment by NealSun96** [comment]
@marouane97 check out files changed, I think u just tabbed the entire page lol. Wanna tab them back so we can have a better diff?

**Comment by marouane97** [comment]
all changes are in the login controller, ill keep the tabbing this in mind next time

File: dova/www/js/controllers.js [hunk 1]
```diff
@@ -1,68 +1,98 @@
...
+    }
+
+    $scope.saveData = function(){
+        window.localStorage.setItem("apiKey", $rootScope.api_auth);
+    }
+
+    //if user data is found, it fills the fields with it and calls login
+    $scope.isLoggedIn = function (){
+        if (window.localStorage.getItem("apiKey") !== null) {
+            $rootScope.api_auth = window.localStorage.getItem("apiKey");
```
Comment (NealSun96) [inline dova/www/js/controllers.js]: @marouane97 You forgot to assign it back :D

**Comment by NealSun96** [comment]
@marouane97 The thing is, we never know if we accidentally changed anything because we don't have automated tests. So I reverted back and checked; everything's fine except for that assign part.
Let's merge this

**Pull request merged**